pkg/registry/resolver.go

```diff
@@ -0,0 +1,54 @@
+package api
+
+import (
+	api "github.com/openshift/ci-tools/pkg/api"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+// Registry will hold all the registry information needed to convert between the
+// user provided configs referencing the registry and the internal, complete
+// represenation
+type Registry struct{}
+
+func (r Registry) Resolve(config api.MultiStageTestConfiguration) (types.TestFlow, []error) {
+	var resolveErrors []error
+	expandedFlow := types.TestFlow{}
+	for _, external := range config.Pre {
+		newStep, errs := toInternal(external)
+		if errs != nil {
+			resolveErrors = append(resolveErrors, errs...)
+			continue
+		}
+		expandedFlow.Pre = append(expandedFlow.Pre, newStep)
+	}
+	for _, external := range config.Test {
+		newStep, errs := toInternal(external)
+		if errs != nil {
+			resolveErrors = append(resolveErrors, errs...)
+			continue
+		}
+		expandedFlow.Test = append(expandedFlow.Test, newStep)
+	}
+	for _, external := range config.Post {
+		newStep, errs := toInternal(external)
+		if errs != nil {
+			resolveErrors = append(resolveErrors, errs...)
+			continue
+		}
+		expandedFlow.Test = append(expandedFlow.Test, newStep)
+	}
+	if resolveErrors != nil {
+		return types.TestFlow{}, resolveErrors
+	}
+	return expandedFlow, nil
+}
+
+func toInternal(input api.TestStep) (types.TestStep, []error) {
```

Comment:
nit: since this cannot error we can omit that from the signature

It is still OK to have the error in the signature for `Resolve()` since we expect it to be possible to return an error in the future.